Automattic/woocommerce-subscriptions-core
# Highlight overdue/late payments in admin subscriptions list

Fixes #260

## Description

Occasionally subscriptions can show as `Active` status when payment is overdue, in particular with offsite billing (e.g. WCPay Subscriptions using Stripe Billing). 

The goal of this PR is to highlight active subscriptions that may have overdue or missed payment issues. 

The existing `i` info button / tooltip component now shows red if payment is overdue. The tooltip message reads: `Subscription payment overdue`. 

This could help merchants take action, e.g. recover payment or suspend. 

Note this is a low-risk tweak to help mitigate this situation with minimal change to the existing UI. This issue is relatively rare in WCPay Subscriptions (estimated affecting 1-5% of merchants), so it's not worth making any more extensive changes right now.

Other options were considered, and could be improvements in future:

- Use a more modern icon (e.g. Gutenberg component).
- Add a new column for complex gateway-related info. Currently we're using a tooltip for two things (see below). This kind of info may be useful elsewhere in UI and/or for filtering.
   1. Indicate renewals are processed offsite – e.g. PayPal or WCPay Subscriptions.
   2. Indicate payment may be overdue (as of this PR). 
- Use a new, specific status when subscriptions are overdue for payment.
![Screenshot 2023-01-19 at 2 19 40 PM](https://user-images.githubusercontent.com/4167300/213333165-4c4141bb-6fc0-40b5-bf2c-bf7840049983.png)


Above screenshot shows both info (1 offsite renewal) and (2 overdue payment). Red icon is only used for overdue, and overdue payment message is displayed first and in bold, to disambiguate & highlight most important issue. When payment is overdue, the renewal mechanism (offsite vs tokenised onsite) is important info, so deliberately including that in tooltip when applicable.

## How to test this PR
1. Ensure you have some subscription products. Can use WC Subscriptions or WCPay Subscriptions. Tip: use daily subscription term so you have more renewal events to test with over time :) 
4. Purchase a subscription or set up a subscription. 
5. Prevent payment on renewal, or simulate this, for example:
   - Manual subscription - no auto renewal, will go overdue on renewal date.
   - Hack subscription next payment date in the past - set `_schedule_next_payment` post meta (aka subscription meta).
6. Go to `wp-admin > Dashboard > Subscriptions`. Any overdue subscriptions should show red icon.

## Product impact


- n/a  Added deprecated functions, hooks or classes to the [spreadsheet](https://docs.google.com/spreadsheets/d/1xw9xszcPMnWsp4C8OKZMsLzZob7tOmWT7qMqmEIq314/edit#gid=0)
